### PR TITLE
feat: add set custom primary colors method in initialize flow

### PR DESCRIPTION
**Pull request by andrey-canon**
**Description:**

This allows to set css custom properties, in this case just primary colors, in order to take advantage of this changes https://github.com/eduNEXT/paragon/pull/1

## How to test
1. install this paragon version https://github.com/openedx/paragon/tree/alpha
2. set mfe api config https://github.com/openedx/frontend-platform/pull/335
3. Add to your lms config the following:
```
    "ENABLE_MFE_CONFIG_API": true,
    "MFE_CONFIG": {
        "CUSTOM_PRIMARY_COLORS": {
            "pgn-color-primary-base": "#AA0000"
        }
    },
```
Result will be something like this
image

### CODE REVIEW

**Comment by felipemontoya** [comment]
As we talked via other channels, I think this is great and it would be even better if this allowed for the optional inclusion of a css variables file.

This way we can continue developing the ways in which to get you the css-variables without the need for a new deployment everytime.

File: src/config.js [hunk 1]
```diff
@@ -203,4 +204,5 @@ export function ensureConfig(keys, requester = 'unspecified application code') {
  * @property {string} MFE_CONFIG_API_URL
  * @property {string} APP_ID
  * @property {string} SUPPORT_URL
+ * @property {string} CUSTOM_PRIMARY_COLORS
```
Comment (dcoa) [inline src/config.js:207]: This should be  an `object` instead of a string property

File: src/initialize.js [hunk 2]
```diff
@@ -157,6 +159,41 @@ export async function runtimeConfig() {
...
 }
 
+/*
+ * Set custom colors based on the config content.
+ * This method allows to change primary colors and its levels on runtime,
+ * if a specific level is already in the configuration that level will have
+ * priority otherwise the level will be calculated based on primary color by
+ * using the mix function.
+ */
+export async function setCustomPrimaryColors() {
```
Comment (dcoa) [inline src/initialize.js:169]: This function doesn't have any process `async` then you should remove this world

File: src/initialize.js [hunk 3]
```diff
@@ -254,6 +291,7 @@ export async function initialize({
     // Configuration
     await handlers.config();
     await runtimeConfig();
+    await setCustomPrimaryColors();
```
Comment (dcoa) [inline src/initialize.js:294]: `Await` is not necessary here because the function doesn't return a promise.

**Review by dcoa — APPROVED** [review]

**Review by johanv26 — APPROVED** [review]

**Pull request merged**